diagram_analysis/diagram_generator.py

```diff
@@ -34,6 +40,8 @@ def __init__(self, repo_location, temp_folder, repo_name, output_dir, depth_leve
         self.meta_agent: MetaAgent | None = None
         self.meta_context = None
         self.depth_level = depth_level
+        self._monitoring_agents: dict[str, CodeBoardingAgent | None] = {}
```

Comment:
Hmm, why are we adding `None` values to the dictionary. Wouldn't it make more sense if they were not in the dictionary if they were `None`? 
I'd like this to be:
```suggestion
        self._monitoring_agents: dict[str, CodeBoardingAgent] = {}
```
we'd need to change the callers to handle this accordingly.